src/frequenz/sdk/model/_manager.py

```diff
@@ -0,0 +1,366 @@
+            NotImplementedError: if the method is not implemented in the subclass.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def _build_path(self, key: Any) -> str:
+        """Build the model path for the given key.
+
+        Args:
+            key: the key for which to build the model path.
+
+        Returns:
+            the model path.  # noqa: DAR202
+
+        Raises:
+            NotImplementedError: if the method is not implemented in the subclass.
+        """
+        raise NotImplementedError
+
+    def get_models_config(self) -> dict[Any, str]:
+        """Get the models configuration.
+
+        Returns:
+            the model keys to model paths mapping.
+        """
+        allowed_keys = self._get_allowed_keys()
+        assert allowed_keys, "Allowed keys must be set."
+        return {key: self._build_path(key) for key in allowed_keys}
+
+
+@dataclass
+class SingleModelRepository(ModelRepository):
+    """Repository to manage a single model."""
+
+    model_path: str
+    """The path to a single model file."""
+
+    def _get_allowed_keys(self) -> list[Any]:
+        """Get the allowed keys for the single model repository.
+
+        Returns:
+            the allowed single key.
+        """
+        return ["Single"]
+
+    def _build_path(self, key: Any) -> str:
+        """Build the model path for the given key.
+
+        Args:
+            key: the key for which to build the model path.
+
+        Returns:
+            the model path.
+
+        Raises:
+            AssertionError: if the given key is not allowed.
+        """
+        assert key in self._get_allowed_keys(), f"Key {key} is not allowed."
+        return self.model_path
+
+
+@dataclass
+class WeekdayModelRepository(ModelRepository):
+    """Repository to manage weekday-based models."""
+
+    directory: str
+    """The base directory where the weekday-based models are stored."""
+
+    file_name: str
+    """The name of the file containing the model."""
+
+    def _get_allowed_keys(self) -> list[Any]:
+        """Get the allowed keys for the weekday-based models repository.
+
+        Returns:
+            the allowed keys for the weekday-based models repository.
+        """
+        return ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+
+    def _build_path(self, key: str) -> str:
+        """Build the model path for the given key.
+
+        Args:
+            key: the key for which to build the model path.
+
+        Returns:
+            The model path.
+
+        Raises:
+            AssertionError: if the directory, file name, or the given key is
+                not set or allowed.
+        """
+        assert self.directory, "Directory must be set."
+        assert self.file_name, "File name must be set."
+        assert key in self._get_allowed_keys(), f"Key {key} is not allowed."
+        return f"{self.directory}/{key}/{self.file_name}"
+
+
+@dataclass
+class IntervalModelRepository(ModelRepository):
```

Comment:
Maybe `MultiModels` or `EnumeratedModels`? Because there is no real interval involved here, just a number of contiguous integers.

Reply:
Then I think `EnumeratedModel` is a good fit because `MultiModels` is more generic